segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/cache/CacheMetrics.java

```diff
@@ -10,28 +10,41 @@
 package io.pravega.segmentstore.storage.cache;
 
 import io.pravega.shared.MetricsNames;
-import io.pravega.shared.metrics.DynamicLogger;
+import io.pravega.shared.metrics.Counter;
 import io.pravega.shared.metrics.MetricsProvider;
+import io.pravega.shared.metrics.StatsLogger;
 
 /**
- * Metrics for {@link CacheStorage}
+ * Metrics for {@link DirectMemoryCache}.
  */
-final class CacheMetrics {
-    private static final DynamicLogger DYNAMIC_LOGGER = MetricsProvider.getDynamicLogger();
+final class CacheMetrics implements AutoCloseable {
```

Comment:
Is this being properly closed everywhere? (This is hard to check outside of an IDE)

Reply:
Yes. It's only used in `DirectMemoryCache` and I added a `metrics.close` when closing it.